networkd/network_server.ml

```diff
@@ -833,6 +859,16 @@ module Bridge = struct
 				(String.concat ", " (List.map (fun (name, _) -> name) config))
 				(if conservative then " (best effort)" else "");
 			List.iter (function (bridge_name, ({ports; vlan; bridge_mac; other_config; _} as c)) ->
+				(* Fail PIF.plug with exception Vlan_in_use if VLAN is in use by kernel and unknown to XAPI *)
+				begin match c.vlan with
+				| None -> ()
+				| Some (parent, id) ->
+					if Xstringext.String.has_substr dbg "PIF.plug" then begin
```

Comment:
Try something like this: https://github.com/robhoes/xcp-networkd/commit/4127f96bcb507da1b6282974e361ab466d7e3d54.